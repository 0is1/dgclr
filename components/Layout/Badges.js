// @flow
import React from 'react';
import { Badge } from 'rebass';
import { getRandomKey } from 'helpers/utils';
import type { Node } from 'react';

type Props = {
  ratings: Array<string>,
  tiny?: boolean,
};

const LayoutRatingBadges = ({ ratings, tiny }: Props) => {
  const badges: Array<?Node> = ratings.map((rating) => {
    if (rating) {
      return tiny ? (
        <Badge mx="8px" py={0} key={getRandomKey()}>
          {rating}
        </Badge>
      ) : (
        <Badge key={getRandomKey()}>{rating}</Badge>
      );
    }
    return null;
  });
  return badges;
};

LayoutRatingBadges.defaultProps = {
  tiny: false,
};

export default LayoutRatingBadges;
