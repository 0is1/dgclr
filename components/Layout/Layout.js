// @flow

import React from 'react';
import { Badge, Box, Flex } from 'rebass';
import { withTranslation } from 'lib/i18n';
import { getRandomKey } from 'helpers/utils';
import CourseStyles from 'components/Course/Course.styles';
import BaseStyles from 'components/Container/Container.styles';
import type { Layout as LayoutType } from 'lib/types';

type Props = {
  active: Boolean,
  layout: LayoutType,
  t: Function,
};

const { Strong } = CourseStyles;
const { BaseText, OL, LI } = BaseStyles;

const Layout = ({ active, layout, t }: Props) => {
  if (!active) return null;
  const { holes } = layout;
  const holeData = holes.map(hole => (
    <LI key={getRandomKey()}>
      <Flex>
        <Box width="70px">
          {!!hole.par && (
            <BaseText>
              <Strong>{t('course:par')}</Strong>
              {` ${hole.par}`}
            </BaseText>
          )}
        </Box>
        <Box width="140px">
          {!!hole.length.meter && (
            <BaseText>
              <Strong>{t('course:length')}</Strong>
              {` ${hole.length.meter}m`}
            </BaseText>
          )}
        </Box>
      </Flex>
    </LI>
  ));
  return (
    <Box width="100%" mt={2} p={2}>
      {layout.rating && <Badge mx="1rem">{layout.rating}</Badge>}
      {layout.holeCount && (
        <BaseText mt={0}>
          <Strong>{t('course:hole-count')}</Strong>
          {`${layout.holeCount}`}
        </BaseText>
      )}
      {layout.holeAverageLength && (
        <BaseText>
          <Strong>{t('course:hole-average-length')}</Strong>
          {`${layout.holeAverageLength.meter}m`}
        </BaseText>
      )}
      {layout.totalPar && (
        <BaseText>
          <Strong>{t('course:course-par')}</Strong>
          {`${layout.totalPar}`}
        </BaseText>
      )}
      {holeData && holeData.length > 0 && (
        <>
          <BaseText>
            <Strong>{t('course:course-holes')}</Strong>
          </BaseText>
          <OL>{holeData}</OL>
        </>
      )}
    </Box>
  );
};

export default withTranslation(['course'])(Layout);
