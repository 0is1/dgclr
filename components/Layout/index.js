import LayoutComponent from './Layout';

export default LayoutComponent;
