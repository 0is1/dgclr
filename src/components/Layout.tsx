import React from "react";
import Head from "next/head";
import { useTranslation } from "next-i18next";
import Footer from "./Footer";
import styles from "../styles/Home.module.css";

type Props = {
  children: React.ReactNode;
};

const Layout = ({ children }: Props) => {
  const { t } = useTranslation(["common"]);
  return (
    <>
      <div className={styles.container}>
        <Head>
          <title>{t("common:title")}</title>
          <meta name="description" content={t("common:sub_title")} />
          <link rel="icon" href="/favicon.ico" />
        </Head>
        <main>{children}</main>
      </div>
      <Footer />
    </>
  );
};

export default Layout;
