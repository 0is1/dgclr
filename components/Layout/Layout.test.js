/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';
import { Badge } from 'rebass';
import { mount } from 'enzyme';
import mockCoursesData from 'components/SearchContainer/mock/courses.mock';
import Layout from './Layout';

const defaultProps = {
  active: true,
  layout: mockCoursesData[1].layouts[0],
  t: jest.fn(),
};

describe('Layout component', () => {
  it('Match snapshot', () => {
    const component = renderer.create(<Layout {...defaultProps} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Renders correctly with active false', () => {
    const enzymeWrapper = mount(<Layout active={false} />);
    expect(enzymeWrapper).toBeDefined();
    expect(enzymeWrapper.find(Badge).length).toEqual(0);
  });
  it('Renders correctly', () => {
    const enzymeWrapper = mount(<Layout {...defaultProps} />);
    expect(enzymeWrapper).toBeDefined();
    expect(enzymeWrapper.find(Badge).length).toEqual(1);
    expect(enzymeWrapper.find(Badge).text()).toEqual('AA1');
  });
});
