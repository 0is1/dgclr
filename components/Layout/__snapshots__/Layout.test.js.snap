// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component Match snapshot 1`] = `
<div
  className="sc-bdVaJa AVsQP"
>
  <div
    className="sc-fAjcbJ xmqIb"
  >
    AA1
  </div>
  <div
    className="sc-gqjmRU Containerstyles__BaseText-sc-1kiz1wt-2 bKftYv"
  >
    <span
      className="Coursestyles__Strong-sc-170ivnh-7 PWqLP"
    >
      Väylien määrä: 
    </span>
    18
  </div>
  <div
    className="sc-gqjmRU Containerstyles__BaseText-sc-1kiz1wt-2 idtCAR"
  >
    <span
      className="Coursestyles__Strong-sc-170ivnh-7 PWqLP"
    >
      Radan par: 
    </span>
    57
  </div>
  <div
    className="sc-gqjmRU Containerstyles__BaseText-sc-1kiz1wt-2 idtCAR"
  >
    <span
      className="Coursestyles__Strong-sc-170ivnh-7 PWqLP"
    >
      Radan väylät: 
    </span>
  </div>
  <ol
    className="Containerstyles__OL-sc-1kiz1wt-5 hBppxg"
  >
    <li
      className="Containerstyles__LI-sc-1kiz1wt-6 jFNgTK"
    >
      <div
        className="sc-bwzfXH hKiLMS sc-bdVaJa iHZvIS"
      >
        <div
          className="sc-bdVaJa bopFKC"
        >
          <div
            className="sc-gqjmRU Containerstyles__BaseText-sc-1kiz1wt-2 idtCAR"
          >
            <span
              className="Coursestyles__Strong-sc-170ivnh-7 PWqLP"
            >
              Par: 
            </span>
             3
          </div>
        </div>
        <div
          className="sc-bdVaJa dYmHol"
        >
          <div
            className="sc-gqjmRU Containerstyles__BaseText-sc-1kiz1wt-2 idtCAR"
          >
            <span
              className="Coursestyles__Strong-sc-170ivnh-7 PWqLP"
            >
              Pituus: 
            </span>
             132m
          </div>
        </div>
      </div>
    </li>
    <li
      className="Containerstyles__LI-sc-1kiz1wt-6 jFNgTK"
    >
      <div
        className="sc-bwzfXH hKiLMS sc-bdVaJa iHZvIS"
      >
        <div
          className="sc-bdVaJa bopFKC"
        >
          <div
            className="sc-gqjmRU Containerstyles__BaseText-sc-1kiz1wt-2 idtCAR"
          >
            <span
              className="Coursestyles__Strong-sc-170ivnh-7 PWqLP"
            >
              Par: 
            </span>
             3
          </div>
        </div>
        <div
          className="sc-bdVaJa dYmHol"
        >
          <div
            className="sc-gqjmRU Containerstyles__BaseText-sc-1kiz1wt-2 idtCAR"
          >
            <span
              className="Coursestyles__Strong-sc-170ivnh-7 PWqLP"
            >
              Pituus: 
            </span>
             56m
          </div>
        </div>
      </div>
    </li>
    <li
      className="Containerstyles__LI-sc-1kiz1wt-6 jFNgTK"
    >
      <div
        className="sc-bwzfXH hKiLMS sc-bdVaJa iHZvIS"
      >
        <div
          className="sc-bdVaJa bopFKC"
        >
          <div
            className="sc-gqjmRU Containerstyles__BaseText-sc-1kiz1wt-2 idtCAR"
          >
            <span
              className="Coursestyles__Strong-sc-170ivnh-7 PWqLP"
            >
              Par: 
            </span>
             4
          </div>
        </div>
        <div
          className="sc-bdVaJa dYmHol"
        >
          <div
            className="sc-gqjmRU Containerstyles__BaseText-sc-1kiz1wt-2 idtCAR"
          >
            <span
              className="Coursestyles__Strong-sc-170ivnh-7 PWqLP"
            >
              Pituus: 
            </span>
             180m
          </div>
        </div>
      </div>
    </li>
    <li
      className="Containerstyles__LI-sc-1kiz1wt-6 jFNgTK"
    >
      <div
        className="sc-bwzfXH hKiLMS sc-bdVaJa iHZvIS"
      >
        <div
          className="sc-bdVaJa bopFKC"
        >
          <div
            className="sc-gqjmRU Containerstyles__BaseText-sc-1kiz1wt-2 idtCAR"
          >
            <span
              className="Coursestyles__Strong-sc-170ivnh-7 PWqLP"
            >
              Par: 
            </span>
             3
          </div>
        </div>
        <div
          className="sc-bdVaJa dYmHol"
        >
          <div
            className="sc-gqjmRU Containerstyles__BaseText-sc-1kiz1wt-2 idtCAR"
          >
            <span
              className="Coursestyles__Strong-sc-170ivnh-7 PWqLP"
            >
              Pituus: 
            </span>
             74m
          </div>
        </div>
      </div>
    </li>
    <li
      className="Containerstyles__LI-sc-1kiz1wt-6 jFNgTK"
    >
      <div
        className="sc-bwzfXH hKiLMS sc-bdVaJa iHZvIS"
      >
        <div
          className="sc-bdVaJa bopFKC"
        >
          <div
            className="sc-gqjmRU Containerstyles__BaseText-sc-1kiz1wt-2 idtCAR"
          >
            <span
              className="Coursestyles__Strong-sc-170ivnh-7 PWqLP"
            >
              Par: 
            </span>
             4
          </div>
        </div>
        <div
          className="sc-bdVaJa dYmHol"
        >
          <div
            className="sc-gqjmRU Containerstyles__BaseText-sc-1kiz1wt-2 idtCAR"
          >
            <span
              className="Coursestyles__Strong-sc-170ivnh-7 PWqLP"
            >
              Pituus: 
            </span>
             162m
          </div>
        </div>
      </div>
    </li>
    <li
      className="Containerstyles__LI-sc-1kiz1wt-6 jFNgTK"
    >
      <div
        className="sc-bwzfXH hKiLMS sc-bdVaJa iHZvIS"
      >
        <div
          className="sc-bdVaJa bopFKC"
        >
          <div
            className="sc-gqjmRU Containerstyles__BaseText-sc-1kiz1wt-2 idtCAR"
          >
            <span
              className="Coursestyles__Strong-sc-170ivnh-7 PWqLP"
            >
              Par: 
            </span>
             3
          </div>
        </div>
        <div
          className="sc-bdVaJa dYmHol"
        >
          <div
            className="sc-gqjmRU Containerstyles__BaseText-sc-1kiz1wt-2 idtCAR"
          >
            <span
              className="Coursestyles__Strong-sc-170ivnh-7 PWqLP"
            >
              Pituus: 
            </span>
             84m
          </div>
        </div>
      </div>
    </li>
    <li
      className="Containerstyles__LI-sc-1kiz1wt-6 jFNgTK"
    >
      <div
        className="sc-bwzfXH hKiLMS sc-bdVaJa iHZvIS"
      >
        <div
          className="sc-bdVaJa bopFKC"
        >
          <div
            className="sc-gqjmRU Containerstyles__BaseText-sc-1kiz1wt-2 idtCAR"
          >
            <span
              className="Coursestyles__Strong-sc-170ivnh-7 PWqLP"
            >
              Par: 
            </span>
             3
          </div>
        </div>
        <div
          className="sc-bdVaJa dYmHol"
        >
          <div
            className="sc-gqjmRU Containerstyles__BaseText-sc-1kiz1wt-2 idtCAR"
          >
            <span
              className="Coursestyles__Strong-sc-170ivnh-7 PWqLP"
            >
              Pituus: 
            </span>
             94m
          </div>
        </div>
      </div>
    </li>
    <li
      className="Containerstyles__LI-sc-1kiz1wt-6 jFNgTK"
    >
      <div
        className="sc-bwzfXH hKiLMS sc-bdVaJa iHZvIS"
      >
        <div
          className="sc-bdVaJa bopFKC"
        >
          <div
            className="sc-gqjmRU Containerstyles__BaseText-sc-1kiz1wt-2 idtCAR"
          >
            <span
              className="Coursestyles__Strong-sc-170ivnh-7 PWqLP"
            >
              Par: 
            </span>
             3
          </div>
        </div>
        <div
          className="sc-bdVaJa dYmHol"
        >
          <div
            className="sc-gqjmRU Containerstyles__BaseText-sc-1kiz1wt-2 idtCAR"
          >
            <span
              className="Coursestyles__Strong-sc-170ivnh-7 PWqLP"
            >
              Pituus: 
            </span>
             90m
          </div>
        </div>
      </div>
    </li>
    <li
      className="Containerstyles__LI-sc-1kiz1wt-6 jFNgTK"
    >
      <div
        className="sc-bwzfXH hKiLMS sc-bdVaJa iHZvIS"
      >
        <div
          className="sc-bdVaJa bopFKC"
        >
          <div
            className="sc-gqjmRU Containerstyles__BaseText-sc-1kiz1wt-2 idtCAR"
          >
            <span
              className="Coursestyles__Strong-sc-170ivnh-7 PWqLP"
            >
              Par: 
            </span>
             3
          </div>
        </div>
        <div
          className="sc-bdVaJa dYmHol"
        >
          <div
            className="sc-gqjmRU Containerstyles__BaseText-sc-1kiz1wt-2 idtCAR"
          >
            <span
              className="Coursestyles__Strong-sc-170ivnh-7 PWqLP"
            >
              Pituus: 
            </span>
             77m
          </div>
        </div>
      </div>
    </li>
    <li
      className="Containerstyles__LI-sc-1kiz1wt-6 jFNgTK"
    >
      <div
        className="sc-bwzfXH hKiLMS sc-bdVaJa iHZvIS"
      >
        <div
          className="sc-bdVaJa bopFKC"
        >
          <div
            className="sc-gqjmRU Containerstyles__BaseText-sc-1kiz1wt-2 idtCAR"
          >
            <span
              className="Coursestyles__Strong-sc-170ivnh-7 PWqLP"
            >
              Par: 
            </span>
             3
          </div>
        </div>
        <div
          className="sc-bdVaJa dYmHol"
        >
          <div
            className="sc-gqjmRU Containerstyles__BaseText-sc-1kiz1wt-2 idtCAR"
          >
            <span
              className="Coursestyles__Strong-sc-170ivnh-7 PWqLP"
            >
              Pituus: 
            </span>
             120m
          </div>
        </div>
      </div>
    </li>
    <li
      className="Containerstyles__LI-sc-1kiz1wt-6 jFNgTK"
    >
      <div
        className="sc-bwzfXH hKiLMS sc-bdVaJa iHZvIS"
      >
        <div
          className="sc-bdVaJa bopFKC"
        >
          <div
            className="sc-gqjmRU Containerstyles__BaseText-sc-1kiz1wt-2 idtCAR"
          >
            <span
              className="Coursestyles__Strong-sc-170ivnh-7 PWqLP"
            >
              Par: 
            </span>
             4
          </div>
        </div>
        <div
          className="sc-bdVaJa dYmHol"
        >
          <div
            className="sc-gqjmRU Containerstyles__BaseText-sc-1kiz1wt-2 idtCAR"
          >
            <span
              className="Coursestyles__Strong-sc-170ivnh-7 PWqLP"
            >
              Pituus: 
            </span>
             213m
          </div>
        </div>
      </div>
    </li>
    <li
      className="Containerstyles__LI-sc-1kiz1wt-6 jFNgTK"
    >
      <div
        className="sc-bwzfXH hKiLMS sc-bdVaJa iHZvIS"
      >
        <div
          className="sc-bdVaJa bopFKC"
        >
          <div
            className="sc-gqjmRU Containerstyles__BaseText-sc-1kiz1wt-2 idtCAR"
          >
            <span
              className="Coursestyles__Strong-sc-170ivnh-7 PWqLP"
            >
              Par: 
            </span>
             3
          </div>
        </div>
        <div
          className="sc-bdVaJa dYmHol"
        >
          <div
            className="sc-gqjmRU Containerstyles__BaseText-sc-1kiz1wt-2 idtCAR"
          >
            <span
              className="Coursestyles__Strong-sc-170ivnh-7 PWqLP"
            >
              Pituus: 
            </span>
             95m
          </div>
        </div>
      </div>
    </li>
    <li
      className="Containerstyles__LI-sc-1kiz1wt-6 jFNgTK"
    >
      <div
        className="sc-bwzfXH hKiLMS sc-bdVaJa iHZvIS"
      >
        <div
          className="sc-bdVaJa bopFKC"
        >
          <div
            className="sc-gqjmRU Containerstyles__BaseText-sc-1kiz1wt-2 idtCAR"
          >
            <span
              className="Coursestyles__Strong-sc-170ivnh-7 PWqLP"
            >
              Par: 
            </span>
             3
          </div>
        </div>
        <div
          className="sc-bdVaJa dYmHol"
        >
          <div
            className="sc-gqjmRU Containerstyles__BaseText-sc-1kiz1wt-2 idtCAR"
          >
            <span
              className="Coursestyles__Strong-sc-170ivnh-7 PWqLP"
            >
              Pituus: 
            </span>
             94m
          </div>
        </div>
      </div>
    </li>
    <li
      className="Containerstyles__LI-sc-1kiz1wt-6 jFNgTK"
    >
      <div
        className="sc-bwzfXH hKiLMS sc-bdVaJa iHZvIS"
      >
        <div
          className="sc-bdVaJa bopFKC"
        >
          <div
            className="sc-gqjmRU Containerstyles__BaseText-sc-1kiz1wt-2 idtCAR"
          >
            <span
              className="Coursestyles__Strong-sc-170ivnh-7 PWqLP"
            >
              Par: 
            </span>
             3
          </div>
        </div>
        <div
          className="sc-bdVaJa dYmHol"
        >
          <div
            className="sc-gqjmRU Containerstyles__BaseText-sc-1kiz1wt-2 idtCAR"
          >
            <span
              className="Coursestyles__Strong-sc-170ivnh-7 PWqLP"
            >
              Pituus: 
            </span>
             123m
          </div>
        </div>
      </div>
    </li>
    <li
      className="Containerstyles__LI-sc-1kiz1wt-6 jFNgTK"
    >
      <div
        className="sc-bwzfXH hKiLMS sc-bdVaJa iHZvIS"
      >
        <div
          className="sc-bdVaJa bopFKC"
        >
          <div
            className="sc-gqjmRU Containerstyles__BaseText-sc-1kiz1wt-2 idtCAR"
          >
            <span
              className="Coursestyles__Strong-sc-170ivnh-7 PWqLP"
            >
              Par: 
            </span>
             3
          </div>
        </div>
        <div
          className="sc-bdVaJa dYmHol"
        >
          <div
            className="sc-gqjmRU Containerstyles__BaseText-sc-1kiz1wt-2 idtCAR"
          >
            <span
              className="Coursestyles__Strong-sc-170ivnh-7 PWqLP"
            >
              Pituus: 
            </span>
             102m
          </div>
        </div>
      </div>
    </li>
    <li
      className="Containerstyles__LI-sc-1kiz1wt-6 jFNgTK"
    >
      <div
        className="sc-bwzfXH hKiLMS sc-bdVaJa iHZvIS"
      >
        <div
          className="sc-bdVaJa bopFKC"
        >
          <div
            className="sc-gqjmRU Containerstyles__BaseText-sc-1kiz1wt-2 idtCAR"
          >
            <span
              className="Coursestyles__Strong-sc-170ivnh-7 PWqLP"
            >
              Par: 
            </span>
             3
          </div>
        </div>
        <div
          className="sc-bdVaJa dYmHol"
        >
          <div
            className="sc-gqjmRU Containerstyles__BaseText-sc-1kiz1wt-2 idtCAR"
          >
            <span
              className="Coursestyles__Strong-sc-170ivnh-7 PWqLP"
            >
              Pituus: 
            </span>
             100m
          </div>
        </div>
      </div>
    </li>
    <li
      className="Containerstyles__LI-sc-1kiz1wt-6 jFNgTK"
    >
      <div
        className="sc-bwzfXH hKiLMS sc-bdVaJa iHZvIS"
      >
        <div
          className="sc-bdVaJa bopFKC"
        >
          <div
            className="sc-gqjmRU Containerstyles__BaseText-sc-1kiz1wt-2 idtCAR"
          >
            <span
              className="Coursestyles__Strong-sc-170ivnh-7 PWqLP"
            >
              Par: 
            </span>
             3
          </div>
        </div>
        <div
          className="sc-bdVaJa dYmHol"
        >
          <div
            className="sc-gqjmRU Containerstyles__BaseText-sc-1kiz1wt-2 idtCAR"
          >
            <span
              className="Coursestyles__Strong-sc-170ivnh-7 PWqLP"
            >
              Pituus: 
            </span>
             124m
          </div>
        </div>
      </div>
    </li>
    <li
      className="Containerstyles__LI-sc-1kiz1wt-6 jFNgTK"
    >
      <div
        className="sc-bwzfXH hKiLMS sc-bdVaJa iHZvIS"
      >
        <div
          className="sc-bdVaJa bopFKC"
        >
          <div
            className="sc-gqjmRU Containerstyles__BaseText-sc-1kiz1wt-2 idtCAR"
          >
            <span
              className="Coursestyles__Strong-sc-170ivnh-7 PWqLP"
            >
              Par: 
            </span>
             3
          </div>
        </div>
        <div
          className="sc-bdVaJa dYmHol"
        >
          <div
            className="sc-gqjmRU Containerstyles__BaseText-sc-1kiz1wt-2 idtCAR"
          >
            <span
              className="Coursestyles__Strong-sc-170ivnh-7 PWqLP"
            >
              Pituus: 
            </span>
             99m
          </div>
        </div>
      </div>
    </li>
  </ol>
</div>
`;
